fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java

```diff
@@ -627,8 +627,9 @@ private boolean processModifyColumn(ModifyColumnClause alterClause, OlapTable ol
                 if (!col.equals(modColumn)) {
                     typeChanged = true;
                     // TODO:the case where columnPos is not empty has not been considered
-                    if (columnPos == null && col.getDataType() == PrimitiveType.VARCHAR
-                            && modColumn.getDataType() == PrimitiveType.VARCHAR) {
+                    if (columnPos == null && ((col.getDataType() == PrimitiveType.VARCHAR
+                            && modColumn.getDataType() == PrimitiveType.VARCHAR)
+                            || col.getDataType().isComplexType())) {
```

Comment:
is this correct?what about col is struct and modColumn is array? this condiction will break the original rule

Reply:
maybe also use isSupportSchemaChangeForCharType

Reply:
I write a isSupportSchemaChangeForCharType for this, here just a pass to call checkSchemaChangeAllowed which can check is support or not

Reply:
it's tricky